File: Cyaim.WebSocketServer/Clients/cyaim-websocket-client-java/src/main/java/com/cyaim/websocket/WebSocketClientOptions.java
package com.cyaim.websocket;

/**
 * Options for creating WebSocket client
 * 创建 WebSocket 客户端的选项
 */
public class WebSocketClientOptions {
    /**
     * Whether to validate all methods have corresponding endpoints (default: false)
     * 是否验证所有方法都有对应的端点（默认：false）
     */
    public boolean validateAllMethods = false;

    /**
     * Whether to fetch endpoints immediately or lazily (default: false)
     * 是否立即获取端点或延迟获取（默认：false）
     */
    public boolean lazyLoadEndpoints = false;

    /**
     * Whether to throw exception if endpoint not found (default: true)
     * 如果找不到端点是否抛出异常（默认：true）
     */
    public boolean throwOnEndpointNotFound = true;
}

File: Cyaim.WebSocketServer/Clients/cyaim-websocket-client-python/cyaim_websocket_client/websocket_client_options.py
from enum import Enum

class SerializationProtocol(Enum):
    """Serialization protocol for WebSocket messages"""
    JSON = 0  # JSON protocol (text messages)
    MESSAGEPACK = 1  # MessagePack protocol (binary messages)

class WebSocketClientOptions:
    """Options for creating WebSocket client"""
    
    def __init__(
        self,
        protocol: SerializationProtocol = SerializationProtocol.JSON,
        validate_all_methods: bool = False,
        lazy_load_endpoints: bool = False,
        throw_on_endpoint_not_found: bool = True
    ):
        self.protocol = protocol
        self.validate_all_methods = validate_all_methods
        self.lazy_load_endpoints = lazy_load_endpoints
        self.throw_on_endpoint_not_found = throw_on_endpoint_not_found

